src/asm2wasm.h

```diff
@@ -1370,13 +1387,34 @@ void Asm2WasmBuilder::processAsm(Ref ast) {
         auto* offset = add->right->cast<CallImport>();
         auto tableName = offset->target;
         if (parent->functionTableStarts.find(tableName) == parent->functionTableStarts.end()) return;
-        add->right = parent->builder.makeConst(Literal((int32_t)parent->functionTableStarts[tableName]));
+        if (parent->compactFunctionTable) {
+          add->right = parent->builder.makeConst(Literal((int32_t)0));
+          if (add->left->is<Binary>()) {
+            auto* and_op = add->left->dynCast<Binary>();
+            //this is masking the index, we can eliminate it
+            if (and_op->op == BinaryOp::AndInt32) {
+              add->left = and_op->left; //FIXME figure out which one is the mask operand
+            }
+          }
+        } else {
```

Comment:
I may be missing something, but this seems like a separate optimization, to get rid of the masking? That is, it could work with or without the above shrinking of the emitted table?

If so it may still make sense to combine it under a single option, but we should document it and perhaps change the name.

To be careful here, we should verify the mask size is the size of the table.

Also, we'd probably want to write a helper function to share this code with the other instance later down (also the 2 preceding lines look identical).